packages/components/videoPlayer/UploadVideoModal.tsx

```diff
@@ -0,0 +1,405 @@
+import { SVG } from "../SVG";
+import { PrimaryButton } from "../buttons/PrimaryButton";
+import { FileUploader } from "../fileUploader";
+import { TextInputCustom } from "../inputs/TextInputCustom";
+import { SpacerColumn, SpacerRow } from "../spacer";
+
+interface UploadAlbumModalProps {
+  onClose: () => void;
+  isVisible: boolean;
+}
+const DEFAULT_VIDEO_INFO = {
+  identifier: "",
+  createdBy: "",
+  createdAt: 0,
+  viewCount: 0,
+  like: 0,
+  dislike: 0,
+  viewLastTimestamp: 0,
+  videoMetaInfo: {
+    title: "",
+    description: "",
+    url: "",
+    image: "",
+    duration: 0,
+  },
+};
+
+export const MODAL_WIDTH = 564;
+export const UploadVideoModal: React.FC<UploadAlbumModalProps> = ({
+  onClose,
+  isVisible,
+}) => {
+  const { setToastError, setToastSuccess } = useFeedbacks();
+  const selectedNetworkId = useSelectedNetworkId();
+  const wallet = useSelectedWallet();
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [isUploading, setIsUploading] = useState(false);
+  const selectedWallet = useSelectedWallet();
+  const userId = getUserId(selectedNetworkId, selectedWallet?.address);
+  const userIPFSKey = useSelector(selectNFTStorageAPI);
+  const [videoInfo, setVideoInfo] =
+    useState<VideoInfoWithMeta>(DEFAULT_VIDEO_INFO);
+
+  const onUploadVideoImage = async (files: LocalFileData[]) => {
+    setIsUploading(true);
+    const uploadedFile = await uploadFileToIPFS({
+      userKey: userIPFSKey,
+      file: files[0],
+      networkId: selectedNetworkId,
+      userId,
+    });
+    if (!uploadedFile?.url) {
+      console.error("upload file err : Fail to pin to IPFS");
+      setToastError({
+        title: "File upload failed",
+        message: "Fail to pin to IPFS, please try to Publish again",
+      });
+      return;
+    }
+    setVideoInfo({
+      ...videoInfo,
+      videoMetaInfo: { ...videoInfo.videoMetaInfo, image: uploadedFile.url },
+    });
+    setIsUploading(false);
+  };
+
+  const onUploadVideoFile = async (files: LocalFileData[]) => {
+    setIsUploading(true);
+    const pinataJWTKey =
+      userIPFSKey || (await generateIpfsKey(selectedNetworkId, userId));
+    if (!pinataJWTKey) {
+      console.error("upload file err : No Pinata JWT");
+      setToastError({
+        title: "File upload failed",
+        message: "No Pinata JWT",
+      });
+      return;
+    }
+    const uploadedFile = await uploadFileToIPFS({
+      userKey: userIPFSKey,
+      file: files[0],
+      networkId: selectedNetworkId,
+      userId,
+    });
+    if (!uploadedFile?.url) {
+      console.error("upload file err : Fail to pin to IPFS");
+      setToastError({
+        title: "File upload failed",
+        message: "Fail to pin to IPFS, please try to Publish again",
+      });
+      return;
+    }
+    setVideoInfo((videoInfo) => {
+      return {
+        ...videoInfo,
+        videoMetaInfo: {
+          ...videoInfo.videoMetaInfo,
+          url: uploadedFile.url,
+          duration: 0, //TODO
```

Comment:
For the Music feature, we can get the `duration` from the audio file.
But not with video files.
The video handling in MediaPlayer is quite different: I get the video duration from expo-av `playbackStatus`.
For music, I get the sound duration from the metadata (from the uploaded file).

I will just remove `duration` from this line.

I can't fully test the components from `packages/components/videoPlayer` because the backend won't returns the videos from DB..